src/main/java/at/ac/tuwien/kr/alpha/solver/heuristics/VSIDS.java

```diff
@@ -0,0 +1,229 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+import static at.ac.tuwien.kr.alpha.common.Literals.atomOf;
+import static at.ac.tuwien.kr.alpha.common.Literals.atomToLiteral;
+
+/**
+ * This implementation is inspired by the VSIDS implementation in <a href="https://github.com/potassco/clasp">clasp</a>.
+ * Therefore, for example, decay is not realized by decreasing activity scores with age, but by
+ * steadily increasing the increment added to the score of active atoms
+ * (such that the activity score of older atoms does not have to be changed later).
+ * <p/>
+ * The implementation is simplified in some ways, e.g. it is not possible to specify a frequency in which the activity
+ * increment is updated (which corresponds to decay), but it is updated after every conflict.
+ * <p/>
+ * Reference for VSIDS:
+ * Moskewicz, Matthew W.; Madigan, Conor F.; Zhao, Ying; Zhang, Lintao; Malik, Sharad (2001):
+ * Chaff: engineering an efficient SAT solver.
+ * In: Proceedings of the 38th Design Automation Conference. IEEE, pp. 530–535.
+ * <p/>
+ * Copyright (c) 2018 Siemens AG
+ */
+public class VSIDS implements ActivityBasedBranchingHeuristic {
+	protected static final Logger LOGGER = LoggerFactory.getLogger(VSIDS.class);
+
+	public static final int DEFAULT_SIGN_COUNTER = 0;
+
+	public static final int DEFAULT_DECAY_PERIOD = 1;
+	public static final double DEFAULT_DECAY_FACTOR = 1 / 0.92;
+
+	protected final Assignment assignment;
+	protected final ChoiceManager choiceManager;
+	protected final Random rand;
+
+	protected final HeapOfActiveAtoms heapOfActiveAtoms;
+	protected final Map<Integer, Integer> signCounters = new HashMap<>();
+
+	private final Collection<NoGood> bufferedNoGoods = new ArrayList<>();
+
+	private int nChoicesTrue;
+	private int nChoicesFalse;
+	private int nChoicesRand;
+
+	/**
+	 * Maps rule heads to atoms representing corresponding bodies.
+	 */
+	protected final MultiValuedMap<Integer, Integer> headToBodies = new HashSetValuedHashMap<>();
+
+	protected VSIDS(Assignment assignment, ChoiceManager choiceManager, HeapOfActiveAtoms heapOfActiveAtoms, Random random, MOMs.Strategy momsStrategy) {
+		this.assignment = assignment;
+		this.choiceManager = choiceManager;
+		this.heapOfActiveAtoms = heapOfActiveAtoms;
+		this.heapOfActiveAtoms.setMOMsStrategy(momsStrategy);
+		this.rand = random;
+	}
+
+	public VSIDS(Assignment assignment, ChoiceManager choiceManager, int decayPeriod, double decayFactor, Random random, MOMs.Strategy momsStrategy) {
+		this(assignment, choiceManager, new HeapOfActiveAtoms(decayPeriod, decayFactor, choiceManager), random, momsStrategy);
+	}
+
+	public VSIDS(Assignment assignment, ChoiceManager choiceManager, Random random, MOMs.Strategy momsStrategy) {
+		this(assignment, choiceManager, DEFAULT_DECAY_PERIOD, DEFAULT_DECAY_FACTOR, random, momsStrategy);
+	}
+
+	@Override
+	public void violatedNoGood(NoGood violatedNoGood) {
```

Comment:
Good question. It´s been a while since `BerkMin` and `DependencyDrivenHeuristic` have been developed, and we already know that they have potential for improvement (I just opened #181 for that). That said, the implementations of `violatedNoGood` and `analyzedConflict` do different things (the former pushes the violated nogood to the stack, the latter pushes the learned nogood to the stack), so I suggest to move further analysis of this question to #181.